framework/merchant_api/domain/entities/scraped_product_offer.py

```diff
@@ -19,15 +19,26 @@ class ScrapedProductOffer:
     image: bytes
     image_uri: str
     is_available: bool
-    merchant: str # TODO: hmm...str? or id of merchant? would have to use GetMerchants or create if not found
+    merchant_name: str
     merchant_stockcode: str
     name: str
+    price_difference: float = field(init=False)
     price_now: float
+    price_per_cup: str | None
     price_was: float
+    size : str = field(init=False)
     size_unit: str
     size_value: float
     web_url: str
 
+    # TODO LOOk if can do in method before translates
+    def __post_init__(self):
+        self.name = self.name if self.brand in self.name else f"{self.brand} {self.name}"
```

Comment:
1. Moved self.name. 
Think it was only Coles with the issue. If Woolies has the issue too we can add the logic to their translator method.

2. self.price_difference is calculated so that should stay

3. Moved self.size
IDK why i was calculating it, that value is already being scraped so i am now using that

4. Should logic to sanitise price now & price was to 2 decimals places stay in the post init?
I imagine we want that for all merchants?

Reply:
yeah that all makes sense, happy to leave it as is now